recipes/natural_language_processing/summarizer/app/summarizer.py

```diff
@@ -44,14 +45,30 @@ def chunk_text(text):
         text_list = text_list[chunk_size:]
     return chunks
 
+def read_file(file):
+    file_type = file.type
+    
+    if file_type == "application/pdf":
+        with open(file.name, "wb") as f:
+            f.write(file.getvalue())
+        loader = PyPDFLoader(file.name)
+        pages = loader.load()
+        text = "".join([p.page_content for p in pages]) 
+        os.remove(file.name) 
+    
+    if file_type == "text/plain":
+        text = file.read().decode()   
+    
+    return text
+
 st.title("🔎 Summarizer")
-file = st.file_uploader("Upload file")
+file = st.file_uploader("Upload file",type=[".txt",".pdf"])
```

Comment:
if I upload a text file without the `.txt` extension, it will fail, right? If so, it might be better to document "Upload text file or pdf" and not enforce there, wdyt?

Reply:
I don't love that idea. Mainly because we use `file.type` to determine how we handle opening the file and that value gets determined by the extension. We could, but I think its probably simpler for users to see that they need to add ".txt" to their file names, then it is to manage the open-ended errors that could occur by allowing any file type to be uploaded in the ui.  

What do others think?